net/tcp/Kconfig

```diff
@@ -51,11 +51,45 @@ config NET_TCPURGDATA
 		compiled in. Urgent data (out-of-band data) is a rarely used TCP feature
 		that is very seldom would be required.
 
-config NET_TCP_CONNS
-	int "Number of TCP/IP connections"
+config NET_TCP_PREALLOC_CONNS
+	int "Preallocated TCP/IP connections"
 	default 8
 	---help---
-		Maximum number of TCP/IP connections (all tasks)
+		Number of TCP/IP connections (all tasks).
+		
+		This number of connections will be pre-allocated during system boot.
+		If dynamic connections allocation is enabled, more connections may
+		be allocated at a later time, as the system needs them. Else this
+		will be the maximum number of connections available to the system
+		at all times.
+
+		Set to 0 to disable (and rely only on dynamic allocations).
+
+config NET_TCP_ALLOC_CONNS
+	int "Dynamic TCP/IP connections allocation"
+	default 0
```

Comment:
```suggestion
	default NET_TCP_PREALLOC_CONNS
```

Reply:
@xiaoxiang781216 Why this recommendation?

The default that I used make the system behave exactly as before.  
There are some statically allocated connections, and dynamic allocation is disabled.

Your proposition alters the default behavior.

Furthermore, why the preallocated connections are related to the dyn. allocation batch size?

Reply:
@anchao I disagree.

I think it is greatly beneficial to have the ability to fine tune each protocol independently.

For example, a system may have a finite number of expected UDP connections, so dyn. alloc. is disabled for safety.  
On the other hand, TCP may be unpredictable, so more connections may be needed.

Reply:
> @xiaoxiang781216 Why this recommendation?
> 
> The default that I used make the system behave exactly as before. There are some statically allocated connections, and dynamic allocation is disabled.
> 
> Your proposition alters the default behavior.
> 

Before this change, the dynamical allocation is enabled by default, I think.

> Furthermore, why the preallocated connections are related to the dyn. allocation batch size?

let change 0 to 8.

Reply:
I see that the default was to have dynamic connections disabled:  
https://github.com/apache/nuttx/blob/master/net/Kconfig#L338

Do you insist on enabling it now?  
Maybe setting it to 1 would be safer?

Reply:
Ok, it's fine to keep the behavior same as before.